### DISCUSSION TITLE
Querying nested models using the TS client

### DISCUSSION THREAD

**Discussion by iaingymware** [alirezanet/Gridify#160] [Q&A]
Hello.

I am using the TypeScript client for my frontend & the .net package in my .net web API.

My search is essentially a "fuzzy" search of such, that I will want to filter based on any data entity in my datatable.  I have the following TS  code:

`   const query = new GridifyQueryBuilder()
    .setPage(options.pageIndex + 1)
    .setPageSize(options.pageSize)
    .startGroup()
      .addCondition("name", op.Contains, search)
      .or()
      .addCondition("phone", op.Contains, search)
      .or()
      .addCondition("email", op.Contains, search)
      .or()
      .addCondition("address.line1", op.Contains, search)
      .or()
      .addCondition("address.line2", op.Contains, search)
    .endGroup()
    .build();`

The last two conditions are where I want to search the address child model, the line1 or line2 property.

What is the approach for handling this in my .net method?

Is there a specific way to handle child models within the TS client (I can't see it from the docs).

Thanks

**Comment by alirezanet** [comment]
Hi @iaingymware,
how did you define your mappings in the backend? it is important to have the desired mapping also in the backend in order to support child properties, for example `address.line1` should be mapped like `.AddMap("address.line1", q=> q.Address.Line1)` 

also [this ](https://alirezanet.github.io/Gridify/guide/gridifyMapper#filtering-on-nested-collections)might be useful:

**Comment by iaingymware** [comment]
I haven't done so yet, so I think this question was more asking around the approach on how this should be done.
